src/main/java/org/reactome/server/controller/DoiController.java

```diff
@@ -0,0 +1,53 @@
+package org.reactome.server.controller;
+
+
+import org.reactome.server.graph.domain.result.PathwayResult;
+import org.reactome.server.graph.service.DoiService;
+
+import org.reactome.server.util.DoiPathwayCache;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.Collection;
+
+
+@SuppressWarnings("unused")
+@Controller
+class DoiController {
+
+    private static final String TITLE = "title";
+    private static final String DOIPATHWAYS = "doiPathways";
+
+    private DoiService doiService;
```

Comment:
It is bit confusing to me, why do you have the DoiService and DoiPathwayCache..
Remember that the controllers have to be  as dummy as it could be, and all the logic kept away from it. If you already have one class (e.g DoiPathwayCache) that could give you back the cached data OR prepare and return and you don't need this logic in the controller.

Reply:
ok, will remove it, thank you